pkg/estimator/server/server.go

```diff
@@ -120,9 +124,18 @@ func NewEstimatorServer(
 		es.informerManager.Lister(gvr)
 	}
 
+	registry := frameworkplugins.NewInTreeRegistry()
+	estimateFramework, err := frameworkruntime.NewFramework(registry,
+		frameworkruntime.WithInformerFactory(informerFactory),
```

Comment:
Add kubeClient here, cause you may need to retreive some objects like resource quota or something else. If possile, make 'get kubeClient' as a interface func could be reasonable.

Reply:
In the inteface.go https://github.com/karmada-io/karmada/pull/4534/files#diff-517e6d53713eb677525f7bc765326759bd4d46cf6fa1940e4fabfc814b3ae561

I have Handle interface
```
// Handle provides data and some tools that plugins can use. It is
// passed to the plugin factories at the time of plugin initialization. Plugins
// must store and use this handle to call framework functions.
// We follow the design pattern as kubernetes scheduler framework
type Handle interface {
	SharedInformerFactory() informers.SharedInformerFactory
}
```

If we need KubeClient in the future we can add Kubeclientand provide a withKubeClientOption func like how we pass sharedInformerFactory

Reply:
In our ResourceQuota Plugin, I need the `SharedInformerFactory` and will use the Lister to get ResourceQuota from local indexer. The estimator will cache all ResourceQuota.

Reply:
I add The KubeClient

Reply:
It's not an objection, but we don't need the `KubeClient` for this PR, we can add it anytime when we need it. It's ok to keep it if you think we will use it. :)

Reply:
Sure.